controllers/shifts.js

```diff
@@ -82,9 +98,13 @@ module.exports = {
   },
   removeAvailable: async(req, res) => {
     try {
-      await Shift.findOneAndUpdate({ _id: req.body.shiftID }, {
+      const shiftData = await Shift.findOneAndUpdate({ _id: req.body.shiftID }, {
         $pull: { availability: req.user.userName }
-      })
+      }, { new: true })
+
+      if (shiftData.availability.length == 0 && shiftData.ownerDisplay == false) {

```

Comment:
Just as I was saying, you're having to repeat yourself here because your model is not self-contained.